api/src/main/java/org/openmrs/module/appointments/advice/RecurringAppointmentsAdvice.java

```diff
@@ -29,7 +29,10 @@ public void afterReturning(Object returnValue, Method method, Object[] arguments
                 super.afterReturning(appointment, method, arguments, target);
             }
         } else if (isSingleRecurringAppointmentUpdate(method, returnValue)) {
-            super.afterReturning(returnValue, method, arguments, target);
+            List<Appointment> updatedAppointments = (List<Appointment>) arguments[1];
+            for (Appointment appointment : updatedAppointments) {
+                super.afterReturning(appointment, method, arguments, target);
+            }
```

Comment:
if-condition says single appointment but iterating over a list of updatedAppointments

Reply:
When we update single appointment in a Recurring series we will have two appointments which get updated which is related appointment. So we have 2 appointments for single appointment update